### ISSUE TITLE
We should set LOADABLE_DRIVERS=OFF automatically when remote enabled

### ISSUE THREAD

**Issue by pjaaskel** [pocl/pocl#1297]
This is documented, but easy to miss. Without it, make check fails with

```
symbol lookup error: /home/pjaaskel/src/pocl/build-cpu/lib/CL/libpocl.so.2.12.0: undefined symbol: pocl_remote_setup_peer_mesh
```

**Comment by jansol** [comment]
I would rather just make remote work with LOADABLE_DRIVERS=ON. There's just the question of what to do about that one function that is (currently) called once during init.

**Comment by pjaaskel** [comment]
Yes, of course fixing the root cause would be better. I assumed it's not trivial as there was a workaround documented instead of a fix.

**Comment by jansol** [comment]
It's not trivial since it needs a bit of special casing, but it should not be particularly difficult either. There just was no acute need to spend the time to do said special handling so far.

**Comment by isuruf** [comment]
Maybe I'm misunderstanding things, but can `pocl_remote_setup_peer_mesh` be done in `pocl_remote_init_device_ops`?

**Comment by jansol** [comment]
AIUI `pocl_remote_init_device_ops` is called separately for each device? Meanwhile `pocl_remote_setup_peer_mesh` iterates over *servers* which likely do not have a 1:1 correspondence with devices.

Currently `pocl_remote_setup_peer_mesh` assumes that all servers are set up, but it would not be too tricky to make it work incrementally. For it to work as part of `pocl_remote_init_device_ops` there would also need to be some checks for whether the server corresponding to the current device has already been added to the mesh. I guess simply adding a flag for this to the server data struct would work in the interim.

**Comment by isuruf** [comment]
> pocl_remote_init_device_ops is called separately for each device?

No, it's called per backend. Once for basic, pthreads, cuda, remote, etc.

**Comment by jansol** [comment]
Ah, in that case yes that should indeed Just Work :tm:

**Comment by pjaaskel** [comment]
This seems to be still a problem.
